mpi-proxy-split/mpi-wrappers/mpi_request_wrappers.cpp

```diff
@@ -169,6 +202,43 @@ USER_DEFINED_WRAPPER(int, Waitall, (int) count,
   return retval;
 }
 
+USER_DEFINED_WRAPPER(int, Waitany, (int) count,
+                     (MPI_Request *) array_of_requests, (int *) index,
+                     (MPI_Status *) status)
+{
+  // FIXME: handle when one or more handles are inactive. currently returns
+  // *index = i instead of MPI_UNDEFINED - likely to require a logic rewrite
+  int retval;
+  int flag = 0;
+  bool all_null = true;
+  *index = MPI_UNDEFINED;
+  while (1) {
+    for (int i = 0; i < count; i++) {
+      if (array_of_requests[i] == MPI_REQUEST_NULL) {
+        continue;
+      }
+      all_null = false;
+      retval = MPI_Test_internal(&array_of_requests[i], &flag, status, false);
+      if (retval != MPI_SUCCESS) {
+        return retval;
+      }
+      if (flag) {
+        if (MPI_LOGGING()) {
+          clearPendingRequestFromLog(array_of_requests[i]);
+          REMOVE_OLD_REQUEST(array_of_requests[i]);
+          array_of_requests[i] = MPI_REQUEST_NULL;
+        }
+        *index = i;
+        return retval;
+      }
+    }
+    if (all_null) {
+      return retval;
+    }
+    sleep(1);
```

Comment:
Why need it?

Reply:
I was spacing out the `MPI_Test_internal` calls while testing so the logs wouldn't be cluttered. I can remove it if it's unnecessary.

Reply:
I'm not sure how much impact it can have on performance, sleeping 1 second sounds a lot to me, and we shouldn't do it if it's unnecessary.

If you just want a better log, one option is to use it only for debug build, we shouldn't have it for release build.